api/v1alpha1/consulbackupplan_types.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 Backup Operator Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const ConsulBackupPlanKind = "ConsulDBBackupPlan"
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ConsulBackupPlanSpec defines the desired state of ConsulBackupPlan
+type ConsulBackupPlanSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Schedule in cron format
+	Schedule string `json:"schedule"`
+
+	// +kubebuilder:validation:Minimum=1
+	//
+	ActiveDeadlineSeconds int64 `json:"activeDeadlineSeconds"`
+
+	// +kubebuilder:validation:Minimum=1
+	// Number of backups to keep
+	Retention int64 `json:"retention"`
+
+	// +optional
+	// Environments for the CronJob
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Fully qualifying Consul URI connection string. Environment variables
+	// will be evaluated before usage.
+	URI string `json:"uri"`
```

Comment:
Looking through the code this is not actually a URI,, but rather it is `Address`.
And we should probably introduce a field `Username` and `Password` as optional fields as well.

Reply:
Fair enough, will change.